Sources/Commands/Describe.swift

```diff
@@ -10,85 +10,355 @@
 
 import TSCBasic
 import PackageModel
+import Foundation
+
 
 enum DescribeMode: String {
-    /// JSON format.
+    /// JSON format (guaranteed to be parsable and stable across time).
     case json
-
-    /// Human readable format.
+    /// Human readable format (not guaranteed to be parsable).
     case text
 }
 
+
+/// Emits a textual description of `package` to `stream`, in the format indicated by `mode`.
 func describe(_ package: Package, in mode: DescribeMode, on stream: OutputByteStream) {
+    let desc = DescribedPackage(from: package)
+    let data: Data
     switch mode {
     case .json:
-        stream <<< package.toJSON().toString(prettyPrint: true) <<< "\n"
+        let encoder = JSONEncoder()
```

Comment:
fyi: https://github.com/apple/swift-package-manager/pull/3071 

we can merge yours first if ready and adjust 3071 or the other way around

Reply:
No, go ahead and merge yours (I'm reviewing right now, though it looks is if there's also a test failure).  I'll adjust mine.  I just pushed some updates to also add the target and product membership.

Reply:
ugh, infra issue. kicked it of again. will merge when green

Reply:
Great, thanks!